### PR TITLE
Connection cache fixes

### PR DESCRIPTION
**Pull request by asmodehn**

### CODE REVIEW

File: rocon_python_comms/src/rocon_python_comms/connections.py [hunk 1]
```diff
@@ -237,26 +256,28 @@ class ConnectionCache(object):
...
     method to force a refresh of the basic information for every connection.
-    For detailed information, run 'generate_type_info' and
-    'generate_xmlrpc_info' on the specific connection (this is a more
-    expensive call and you don't want to do this for every connection).
     """
-    __slots__ = ['connections',
-                 '_lookup_node',
-                 '_get_system_state'
-                 ]
+    __slots__ = [
```
Comment (stonier) [inline rocon_python_comms/src/rocon_python_comms/connections.py]: Feel free to drop these slots if they serve no purpose. Should be adhering more strictly to having sphinx docs for class variables and defining them all (even if None) in the init.

**Comment by stonier** [comment]
This pull request is a WIP still right? We'll need to make sure we merge `rocon_msgs`, `rocon_tools`, `rocon_multimaster` and `pyros` simulatenously to prevent breakages.

**Comment by asmodehn** [comment]
Required pull requests : 
https://github.com/robotics-in-concert/rocon_msgs/pull/116

File: rocon_python_comms/src/rocon_python_comms/connections.py [hunk 2]
```diff
@@ -137,42 +156,56 @@ def msg(self):
...
+                    self.type_msg = self.type_info
+                except rostopic.ROSTopicIOException as topic_exc:
+                    rospy.logwarn(topic_exc)
             elif self.type == SERVICE:
-                self.type_info = rosservice.get_service_uri(self.name)
+                try:
+                    self.type_info = rosservice.get_service_uri(self.name)
+                    self.type_msg = rosservice.get_service_type(self.name)
+                except rosservice.ROSServiceIOException as service_exc:
+                    rospy.logwarn(service_exc)
```
Comment (stonier) [inline rocon_python_comms/src/rocon_python_comms/connections.py:180]: Can you prefix this guy, something like:

```
rospy.logwarn("Connection: %s" % service_exc)
```

That's our usual convention as it gets very hard to track down which component is throwing obscure messages without really digging into the system.

File: rocon_python_comms/src/rocon_python_comms/connections.py [hunk 3]
```diff
@@ -137,42 +156,56 @@ def msg(self):
...
-                goal_topic = self.name + '/goal'
-                goal_topic_type = rostopic.get_topic_type(goal_topic)
-                self.type_info = re.sub('ActionGoal$', '', goal_topic_type[0])  # Base type for action
+                try:
+                    goal_topic = self.name + '/goal'
+                    goal_topic_type = rostopic.get_topic_type(goal_topic)
+                    self.type_info = re.sub('ActionGoal$', '', goal_topic_type[0])  # Base type for action
+                    self.type_msg = self.type_info
+                except rostopic.ROSTopicIOException as topic_exc:
+                    rospy.logwarn(topic_exc.msg)
```
Comment (stonier) [inline rocon_python_comms/src/rocon_python_comms/connections.py:188]: Again, prefix.

File: rocon_python_comms/src/rocon_python_comms/connections.py [hunk 4]
```diff
@@ -237,50 +275,37 @@ class ConnectionCache(object):
...
 
     def __init__(self):
-        master = rosgraph.Master(rospy.get_name())
-        self._lookup_node = master.lookupNode
-        self._get_system_state = master.getSystemState
-        self.connections = create_empty_connection_type_dictionary()
+        #: rosgraph.Master: master API instance
+        self.master = rosgraph.Master(rospy.get_name())
 
-    def generate_type_info(self, name):
```
Comment (stonier) [inline rocon_python_comms/src/rocon_python_comms/connections.py:255]: This guy wasn't being used by anyone?

**Comment by stonier** [comment]
Gopher branch is disappearing, will reopen this in a new PR ([dammit](https://github.com/isaacs/github/issues/18)).

**Pull request closed without merging**